### PR TITLE
fix pipeline bugs

### PR DESCRIPTION
**Pull request by shaowenchen**
**What type of PR is this?**
 /kind bug

**What this PR does / why we need it:**
- fix can't create or delete pipeline.
- fix get pipelinerun list error
- feature kapi for devopsproject

### CODE REVIEW

File: pkg/controller/pipeline/pipeline_controller.go [hunk 1]
```diff
@@ -225,30 +224,19 @@ func (c *Controller) syncHandler(key string) error {
...
-				klog.Error(err, fmt.Sprintf("failed to get pipeline %s ", key))
-				return err
-			} else if err != nil && devopsClient.GetDevOpsStatusCode(err) == http.StatusNotFound {
-			} else {
-				if _, err := c.devopsClient.DeleteProjectPipeline(nsName, pipeline.Name); err != nil {
-					klog.Error(err, fmt.Sprintf("failed to delete pipeline %s in devops", key))
-					return err
-				}
+			if _, err := c.devopsClient.DeleteProjectPipeline(nsName, pipeline.Name); err != nil {
+				klog.Error(err, fmt.Sprintf("failed to delete pipeline %s in devops", key))
```
Comment (zryfish) [inline pkg/controller/pipeline/pipeline_controller.go:239]: Should we just ignore the error?  If we just ignore the error, I think the pipeline probably will be stuck there forever.

**Comment by shaowenchen** [comment]
/cc @wansir 

add devopsproject to kapi

**Comment by shaowenchen** [comment]
add pipeline credential for kapi
/cc @wansir

File: pkg/kapis/devops/v1alpha3/handler.go [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+func newDevOpsHandler(devopsClient devopsClient.Interface, k8sclient kubernetes.Interface, ksclient kubesphere.Interface,
+	ksInformers externalversions.SharedInformerFactory,
+	k8sInformers informers.SharedInformerFactory) *devopsHandler {
+
+	return &devopsHandler{
+		devops: devops.NewDevopsOperator(devopsClient, k8sclient, ksclient, ksInformers, k8sInformers),
+	}
+}
+
+// devops
```
Comment (zryfish) [inline pkg/kapis/devops/v1alpha3/handler.go:32]: unfinished comments?

File: pkg/kapis/devops/v1alpha3/handler.go [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+			return
+		}
+		api.HandleBadRequest(response, request, err)
+		return
+	}
+
+	response.WriteEntity(servererr.None)
+}
+
+// pipeline
```
Comment (zryfish) [inline pkg/kapis/devops/v1alpha3/handler.go:141]: unfinished comments

File: pkg/models/devops/devops.go [hunk 4]
```diff
@@ -104,6 +141,111 @@ func convertToHttpParameters(req *http.Request) *devops.HttpParameters {
...
+	project.Annotations[tenantv1alpha1.WorkspaceLabel] = workspace
+	return d.ksclient.DevopsV1alpha3().DevOpsProjects().Update(project)
+}
+
+func (d devopsOperator) ListDevOpsProject(workspace string) (*v1alpha3.DevOpsProjectList, error) {
+	return d.ksclient.DevopsV1alpha3().DevOpsProjects().List(metav1.ListOptions{})
+}
+
+// pipelineobj in crd
+func (d devopsOperator) CreatePipelineObj(workspace string, projectName string, pipeline *v1alpha3.Pipeline) (*v1alpha3.Pipeline, error) {
```
Comment (zryfish) [inline pkg/models/devops/devops.go:167]: Why there is a `Obj` in function name, I think `CreatePipeline` is just fine.

**Pull request merged**